integration/tests/cook/test_cli.py

```diff
@@ -1783,6 +1818,41 @@ def test_cat_binary_file(self):
         self.assertEqual(0, cp.returncode, cp.stderr)
         self.assertEqual(bytes(i for i in range(0, 256)), cp.stdout)
 
+    def test_cat_with_plugin(self):
+        # User defined plugin to print dummy file content from a file
+        with tempfile.NamedTemporaryFile(suffix='.py', delete=True) as temp:
+            plugin_code = """
+class MockResp:
+    def iter_content(self, chunk_size):
+        yield b"Hello"
+        yield b" "
+        yield b"world!"
+
+def dummy_cat_text(_, __, ___):
+    return MockResp()
+"""
+            temp.write(plugin_code.encode())
+            temp.flush()
+            config = {
+                'plugins': {
+                    "cat-plugin": {
+                        "module-name": "does_not_matter",
+                        "path": temp.name,
+                        "function-name": "dummy_cat_text"
+                    }
+                }
+            }
+
+            with cli.temp_config_file(config) as path:
+                flags = f'--config {path} --verbose'
+                cp, uuids = cli.submit('touch file.txt', self.cook_url)
+                self.assertEqual(0, cp.returncode, cp.stderr)
+                cp = cli.wait(uuids, self.cook_url)
+                self.assertEqual(0, cp.returncode, cp.stderr)
+                cp = cli.cat(uuids[0], 'file.txt', self.cook_url, flags=flags)
```

Comment:
Do we even need to submit a job to test this?

Reply:
We need to submit a job to get a uuid to pass into `cat`, and `touch` creates a file to pass into `cat` as well.

Reply:
Do we have to `wait` for the job?

Reply:
I'm not sure if it's completely necessary, but the other `cat` integration tests `wait` until the job is completed, so I thought I would do the same, even if it is just creating a file.

Reply:
Same comment as above